calcom/cal.com
# refactor: Eventtype app context

## What does this PR do?



Stacked off of #9337 

This PR refactors the `eventtypeAppContext` from an array to an object, in order to grab only what is needed.

Fixes # (issue)



## Type of change



- Chore (refactoring code, technical debt, workflow improvements)


## How should this be tested?



- Install event type apps
- Enable & edit the data to ensure they are being saved to the event type metadata
- Create a managed event type and enable some apps
    - In a child event type go to the apps page. Make sure they are enabled and not editable

## Mandatory Tasks


## Checklist



- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes
